atlasdb-api/src/main/java/com/palantir/atlasdb/spi/DerivedConcurrencyConfig.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.spi;
+
+public interface DerivedConcurrencyConfig {
```

Comment:
Discussed offline - pasting from description above to capture the reasoning behind this:

concurrentGetRangesThreadPoolSize() and defaultGetRangesConcurrency() are derived from both runtime and install information. In addition, these values will be snapshot at setup time. Since this can't be optional in the KeyValueServiceRuntimeConfig (as then the Reloadable Config would need to have these as optional fields), and cannot be defined in advance (as the default depends on poolSize (install) and servers (now runtime), I've used a separate interface DerivedConcurrencyConfig to mark these methods that may require a merge.


For now, this seems to be the way to go. Ideally, a goal is to drop `CassandraReloadableKeyValueServiceRuntimeConfig` from the `CassandraAtlasDbFactory` (although I've achieved that in a different way with the latest commit)